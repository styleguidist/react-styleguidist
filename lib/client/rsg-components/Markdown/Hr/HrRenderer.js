import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';

var styles = function styles(_ref) {
  var space = _ref.space,
      color = _ref.color;
  return {
    hr: {
      borderBottom: [[1, color.border, 'solid']],
      marginTop: 0,
      marginBottom: space[2]
    }
  };
};

export var HrRenderer = function HrRenderer(_ref2) {
  var classes = _ref2.classes;
  return /*#__PURE__*/React.createElement("hr", {
    className: classes.hr
  });
};
HrRenderer.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired
};
export default Styled(styles)(HrRenderer);