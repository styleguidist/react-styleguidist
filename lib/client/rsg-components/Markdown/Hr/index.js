export { default } from 'rsg-components/Markdown/Hr/HrRenderer';