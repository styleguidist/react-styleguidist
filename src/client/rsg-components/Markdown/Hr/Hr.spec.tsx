import React from 'react';
import renderer from 'react-test-renderer';

import Hr from './index';

describe('Markdown Hr', () => {
	it('should render a horizontal rule', () => {
		const actual = renderer.create(<Hr />);

		expect(actual.toJSON()).toMatchSnapshot();
	});
});
