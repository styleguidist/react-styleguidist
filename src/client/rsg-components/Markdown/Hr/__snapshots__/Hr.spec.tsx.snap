// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown Hr should render a horizontal rule 1`] = `
<hr
  className="rsg--hr-0"
/>
`;
